js/parts/Point.js

```diff
@@ -165,6 +165,32 @@ Highcharts.Point.prototype = {
 		return point;
 	},
 
+	/**
+	 * Set a value in an object, on the property defined by key. The key
+	 * supports nested properties using dot notation. The function modifies the
+	 * input object and does not make a copy.
+	 *
+	 * @param  {Object} object The object to set the value on.
+	 * @param  {Mixed} value The value to set.
+	 * @param  {String} key Key to the property to set.
+	 *
+	 * @return {Object} The modified object.
+	 */
+	setNestedProperty: function (object, value, key) {
+		var nestedKeys = key.split('.'),
+			nestedRet = object;
+		for (var i = 0; i < nestedKeys.length - 1; ++i) {
```

Comment:
Also `reduce` can be nice to use in cases like `setNestedProperties`:
```javascript
var setNestedProperty = function(object, value, key) {
  var nestedKeys = key.split('.');
  reduce(nestedKeys, function(result, key, i, arr) {
    var isLastKey = arr.length - 1 === i;
    result[key] = (
      isLastKey ?
      value :
      (isObject(result[key]) ? result[key] : {})
    );
    return result[key];
  }, object);
  return object;
}
````

Reply:
The reduce solution is nice! 👍 I'm not sure it is more readable, but it is shorter so adding it. Also, using `isObject` is smart with the strict parameter, since it also catches cases where the objects are arrays. I tested for a performance difference here, and there doesn't seem to be any in practice, since this is not a bottleneck performance area.